Source/WTF/wtf/ParkingLot.cpp

```diff
@@ -376,23 +367,23 @@ void ensureHashtableSize(unsigned numThreads)
 
     // OK, now we resize. First we gather all thread datas from the old hashtable. These thread datas
     // are placed into the vector in queue order.
-    Vector<ThreadData*> threadDatas;
+    Vector<RefPtr<ThreadData>> threadDatas;
     for (Bucket* bucket : reusableBuckets) {
-        while (ThreadData* threadData = bucket->dequeue())
-            threadDatas.append(threadData);
+        while (RefPtr<ThreadData> threadData = bucket->dequeue())
+            threadDatas.append(WTFMove(threadData));
     }
 
     unsigned newSize = numThreads * growthFactor * maxLoadFactor;
-    RELEASE_ASSERT(newSize > oldHashtable->size);
+    RELEASE_ASSERT(newSize > oldHashtable->data.size());
     
-    Hashtable* newHashtable = Hashtable::create(newSize);
+    auto newHashtable = makeUnique<Hashtable>(newSize);
     if (verbose)
-        dataLogForCurrentThread(": created new hashtable: ", RawPointer(newHashtable), "\n");
-    for (ThreadData* threadData : threadDatas) {
+        dataLogForCurrentThread(": created new hashtable: ", RawPointer(newHashtable.get()), "\n");
+    for (auto threadData : threadDatas) {
```

Comment:
Couldn't we use `auto&`? Or would static analysis complain?

Reply:
I was afraid it would complain but I'll give it a shot.